File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/interactors/PlayerInteractor.java
package com.dbottillo.mtgsearchfree.interactors;


import com.dbottillo.mtgsearchfree.model.Player;

import java.util.List;

import rx.Observable;

public interface PlayerInteractor {

    Observable<List<Player>> load();

    Observable<List<Player>> addPlayer(Player player);

    Observable<List<Player>> editPlayer(Player player);

    Observable<List<Player>> editPlayers(List<Player> players);

    Observable<List<Player>> removePlayer(Player player);
}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/interactors/PlayerInteractor.kt
package com.dbottillo.mtgsearchfree.interactors

import com.dbottillo.mtgsearchfree.model.Player

import io.reactivex.Observable

interface PlayerInteractor {

    fun load(): Observable<List<Player>>

    fun addPlayer(): Observable<List<Player>>

    fun editPlayer(player: Player): Observable<List<Player>>

    fun editPlayers(players: List<Player>): Observable<List<Player>>

    fun removePlayer(player: Player): Observable<List<Player>>
}